web/src/main/java/com/google/sps/servlets/SaveStockServlet.java

```diff
@@ -45,22 +49,57 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
     KeyFactory keyFactory = datastore.newKeyFactory().setKind("Stock");
+    ArrayList<String> tikkers = new ArrayList<String>();
+    for (int i = 0; i < tik.size(); i++) {
+      String tikk = tik.get(i).text();
+      tikkers.add(tikk);
+    }
+
 
     for (int i = 0; i < tik.size(); i++) {
+      Key myTiks = datastore.newKeyFactory().setKind("Stock").newKey(tikkers.get(i));
 
-      String tikk = tik.get(i).text();
       String pricee = price.get(i).text().replaceAll("[\\\\$,]", "");
-      double priceDouble = Double.parseDouble(pricee);
+      Double priceDouble = Double.parseDouble(pricee);
 
-      FullEntity taskEntity =
-          Entity.newBuilder(keyFactory.newKey())
-              .set("Tik", tikk)
-              .set("TimeStamp", timeStamp)
+      Entity taskEntity =
+          Entity.newBuilder(datastore.get(myTiks))
```

Comment:
You are still creating a new ticker here. You need to get an existing ticker from datastore.
Here is an example:
```
Query<Entity> query =
        Query.newEntityQueryBuilder().setKind("Stock").build();
    QueryResults<Entity> results = datastore.run(query);

    List<Stock> stocks= new ArrayList<>();
    while (results.hasNext()) {
      Entity datastoreEntity= results.next();

      long id = datastoreEntity.getKey().getId();
      String stockTicker= datastoreEntity.getString("Ticker");
      long price= datastoreEntity.getLong("USD");

      Stock stock = new Stock(id, stockTicker, price);
      stocks.add(stock );

      Entity updatedStockEntity=
            Entity.newBuilder(datastoreEntity).setPrice(1.23).build();
       datastore.updateStock(updatedStockEntity);

    }

    
```

Reply:
I didn't have to use that since I am going to create a new record with the same key, so every time it updates, it will update to the same key.